demos/bubble_shear-goal.py

```diff
@@ -0,0 +1,313 @@
+        return J
+
+    return qoi
+
+
+# We must also define the adaptor function to drive the mesh adaptation process. Here
+# we compute the anisotropic DWR metric which requires us to compute the hessian of the
+# tracer concentration field. We combine each metric in a subinterval by intersection.
+# We will only run two iterations of the fixed point iteration algorithm so we do not
+# ramp up the complexity. ::
+
+
+def adaptor(mesh_seq, solutions, indicators):
+    iteration = mesh_seq.fp_iteration
+    mp = {
+        "dm_plex_metric": {
+            "target_complexity": 1000,
+            "p": 1.0,
+            "h_min": 1e-04,
+            "h_max": 1.0,
+        }
+    }
+
+    metrics = []
+    for i, mesh in enumerate(mesh_seq):
+        c_solutions = solutions["c"]["forward"][i]
+
+        P1_ten = TensorFunctionSpace(mesh, "CG", 1)
+        subinterval_metric = RiemannianMetric(P1_ten)
+
+        for j, c_sol in enumerate(c_solutions):
+            # Recover the Hessian of the forward solution
+            hessian = RiemannianMetric(P1_ten)
+            hessian.compute_hessian(c_sol)
+
+            metric = RiemannianMetric(P1_ten)
+            metric.set_parameters(mp)
+
+            # Compute an anisotropic metric from the error indicator field and Hessian
+            metric.compute_anisotropic_dwr_metric(indicators["c"][i][j], hessian)
+
+            # Combine the metrics from each exported timestep in the subinterval
+            subinterval_metric.intersect(metric)
+        metrics.append(metric)
+
+    # Normalise the metrics in space and time
+    space_time_normalise(metrics, mesh_seq.time_partition, mp)
+
+    complexities = np.zeros(len(mesh_seq))
+    for i, metric in enumerate(metrics):
+        complexities[i] = metric.complexity()
+        mesh_seq[i] = adapt(mesh_seq[i], metric)
+
+    num_dofs = mesh_seq.count_vertices()
+    num_elem = mesh_seq.count_elements()
+    pyrint(f"Fixed point iteration {iteration}:")
+    for i, (complexity, ndofs, nelem) in enumerate(
+        zip(complexities, num_dofs, num_elem)
+    ):
+        pyrint(
+            f"  subinterval {i}, complexity: {complexity:4.0f}"
+            f", dofs: {ndofs:4d}, elements: {nelem:4d}"
+        )
+
+    return True
+
+
+# Finally, we can define the :class:`GoalOrientedMeshSeq` and run the fixed point iteration.
+# For the purposes of this demo, we divide the time interval into 6 subintervals and only
+# run two iterations of the fixed point iteration, which is not enough to reach convergence. ::
+
+# Reduce the cost of the demo during testing
+test = os.environ.get("GOALIE_REGRESSION_TEST") is not None
+n = 50 if not test else 5
+dt = 0.01 if not test else 0.05
+maxiter = 2 if not test else 1  # maximum number of fixed point iterations
+
+num_subintervals = 6
+meshes = [UnitSquareMesh(n, n) for _ in range(num_subintervals)]
+end_time = period / 2
+time_partition = TimePartition(
+    end_time,
+    len(meshes),
```

Comment:
Again, please use `num_subintervals`.

Reply:
Fixed in 262c78a51a99714a4794b55259edf6fc5aea3567.